app/services/concerns/xccdf/rule_identifiers.rb

```diff
@@ -7,18 +7,28 @@ module RuleIdentifiers
 
     included do
       def save_rule_identifiers
-        @rule_identifiers ||= @new_rules.map do |rule|
+        @rule_identifiers ||= new_rule_identifiers.map do |rule|
```

Comment:
This tends to be the issue in all of these - `from_openscap_parser` is a new SQL call.. per record. What I'm doing in here is just making sure we only do these for *new* records, and skip pre-exisiting records.